src/trait_/partial_ord.rs

```diff
@@ -42,10 +43,16 @@ impl TraitImpl for PartialOrd {
 		}
 	}
 
-	fn build_body(&self, traits: &[DeriveTrait], trait_: &DeriveTrait, data: &Data) -> TokenStream {
+	fn build_body(
+		&self,
+		any_bound: bool,
+		traits: &[DeriveTrait],
+		trait_: &DeriveTrait,
+		data: &Data,
+	) -> TokenStream {
 		if data.is_empty(**trait_)
 			|| data.is_incomparable()
-			|| traits.iter().any(|trait_| trait_ == Trait::Ord)
+			|| (!any_bound && traits.iter().any(|trait_| trait_ == Trait::Ord))
 		{
 			TokenStream::new()
 		} else {
```

Comment:
Why do we even hit this branch, shouldn't it be `unreachable`?

Reply:
Ah, you mean why is the whole function being called in the first place?
That's because the caller has no awareness of whats going on, so the function is always called and the implementation has to figure out if it was necessary in the first place or not.